minio/minio
# Add API's for managing bucket quota

This PR allows setting a "hard" or "fifo" quota
restriction at the bucket level. Buckets that
have reached the FIFO quota configured, will
automatically be cleaned up in FIFO manner until
bucket usage drops to configured quota.
If a bucket is configured with a "hard" quota
ceiling, all further writes are disallowed.

## Description


## Motivation and Context


## How to test this PR?
Use the example in madmin/examples/bucket-quota.go to set/get/remove quota. For such buckets, quota enforcement should kick in when hard/fifo quota is set. Background deletion of objects in FIFO configured buckets should happen at configured interval.  This PR hinges on #9208 to be close to accurate on bucket usage in real time, and might need some tweaks after data tracking PR is merged. Other than that it is ready for preliminary review.

## Types of changes

## Checklist:
